Tensile/Common.py

```diff
@@ -1291,33 +1295,36 @@ def GetArchCaps(isaVersion):
 
   return rv
 
-def tryAssembler(isaVersion, options, asmString, debug=False):
+def tryAssembler(isaVersion, asmString, debug=False, *options):
   """
   Try to assemble the asmString for the specified target processor
   Success is defined as assembler returning no error code or stderr/stdout
   """
+  options = list(options)
+  if globalParameters["PrintLevel"] >= 2:
+    debug = True
+
   if isaVersion[0] == 10:
-    options += ' -mwavefrontsize64'
+    options += ['-mwavefrontsize64']
 
-  asmCmd = "%s -x assembler -target amdgcn-amdhsa -mcpu=%s %s -" \
-             % (globalParameters["AssemblerPath"], gfxName(isaVersion), options)
+  args = [globalParameters["AssemblerPath"], '-x', 'assembler',
+          '-target', 'amdgcn-amdhsa',
+          '-mcpu='+gfxName(isaVersion),
+          *options,
+          '-']
 
-  sysCmd = "echo \"%s\" | %s" % (asmString, asmCmd)
+  result = subprocess.run(args, input=asmString.encode(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
Is the try/except not needed anymore?

Reply:
`check_output()` raises an exception if the underlying command returns a non-zero error code.  `run()` by default does not, and just returns an object which has the `returncode` member.